### PR TITLE
[fix] fix 'round' function to round to powers of 10 when scale is negative to meet Spark expectation

### LINKED ISSUE #434: [Bug] "round" function returns original number when scale is negative does not meet Spark expectation
### Component Selection

- [x] Core Engine (Expression eval, Memory, Vector)
- [ ] Connectors / File Formats (Hive, Parquet, etc.)
- [ ] API / Bindings (Python, etc.)
- [ ] Build
- [ ] Other

### Describe the Bug

"round" function returns original number when decimal is negative does not meet Spark expectation.
- https://docs.databricks.com/aws/en/sql/language-manual/functions/round
- https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.round.html

### Reproduction Steps

In Spark, `SELECT round(1, -1)` should return 0. Now it returns 1, `SELECT round(12, -1)` should return 10. Now it returns 12.

### Bolt Version / Commit ID

0be53c2f8bf564ab2029d48b8e9501e56f9a0fa9

### System Configuration

```markdown
- **OS**: (e.g. Ubuntu 22.04, CentOS 7)
- **Compiler**: (e.g. GCC 11, Clang 14)
- **Build Type**: (Debug / Release / RelWithDebInfo)
- **CPU Arch**: (e.g. x86_64 AVX2, ARM64)
- **Framework**: (e.g. Spark 3.3, PrestoDB)
```

### Logs / Stack Trace

```shell

```

### Expected Behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by markjin1990**
### What problem does this PR solve?

Issue Number: close #434

### Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🚀 Performance improvement (optimization)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔨 Refactoring (no logic changes)
- [ ] 🔧 Build/CI or Infrastructure changes
- [ ] 📝 Documentation only

### Description

- Add round function for Spark with `alwaysRoundNegDec` equal to `true`, so that it round to power of 10 when scale < 0. Previously, the registration on the Spark side was referencing the Presto roundFunction and `alwaysRoundNegDec` was default to false (https://github.com/bytedance/bolt/blob/main/bolt/functions/sparksql/registration/RegisterMathMisc.cpp#L35-L41).
  - https://spark.apache.org/docs/latest/api/sql/index.html#round
  - https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.round.html
- Fix `round` function when alwaysRoundNegDec is true but decimal is negative. Previously, nothing is implemented, and it will hit an compilation error.

### Performance Impact

- [x] **No Impact**: This change does not affect the critical path (e.g., build system, doc, error handling).
- [ ] **Positive Impact**: I have run benchmarks.
    <details>
    <summary>Click to view Benchmark Results</summary>

    ```text
    Paste your google-benchmark or TPC-H results here.
    Before: 10.5s
    After:   8.2s  (+20%)
    ```
    </details>
- [ ] **Negative Impact**: Explained below (e.g., trade-off for correctness).

### Release Note

Please describe the changes in this PR

Release Note:

```text
Release Note:
- Fix 'round' function when decimal is negative to meet Spark expectation
```

### Checklist (For Author)

- [x] I have added/updated unit tests (ctest).
- [x] I have verified the code with local build (Release/Debug).
- [x] I have run clang-format / linters.
- [ ] (Optional) I have run Sanitizers (ASAN/TSAN) locally for complex C++ changes.
- [ ] No need to test or manual test.

### Breaking Changes

- [x] No
- [ ] Yes (Description: ...)
    <details>
    <summary>Click to view Breaking Changes</summary>

    ```text
    Breaking Changes:
    - Description of the breaking change.
    - Possible solutions or workarounds.
    - Any other relevant information.
    ```
    </details>

### CODE REVIEW

File: bolt/functions/prestosql/ArithmeticImpl.h [hunk 1]
```diff
@@ -48,8 +49,32 @@ round(const TNum& number, const TDecimals& decimals = 0) {
...
+          sizeof(DecimalUtil::kPowersOfTen[0]);
+      const auto scale = static_cast<uint64_t>(-decimals);
+      if (scale >= kMaxScale) {
+        return static_cast<TNum>(0);
+      }
+
+      const auto scalingFactor = DecimalUtil::kPowersOfTen[scale];
+      int128_t value = static_cast<int128_t>(number);
+      int128_t quotient = value / scalingFactor;
+      int128_t remainder = value % scalingFactor;
```
Comment (zhangxffff) [inline bolt/functions/prestosql/ArithmeticImpl.h:70]: Can we avoid `int128_t` `/` and `%` when `TNum` is not `int128_t`? This would eliminate the significant overhead of `int128_t` arithmetic.

Comment (markjin1990) [reply bolt/functions/prestosql/ArithmeticImpl.h:70]: @zhangxffff Thanks for the comment! Replacing int128_t with int64_t might be risky when scalingFactor is very large (DecimalUtil::kPowersOfTen is defined as int128_t), and `quotient * scalingFactor` can overflow. I think we should prefer correctness over performance in this case.

File: bolt/functions/sparksql/tests/SparkRoundTest.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+  template <typename T>
+  std::vector<std::tuple<T, T>> testRoundIntegralData() {
+    return {{1, 1}, {0, 0}, {-1, -1}};
+  }
+
+  template <typename T>
+  std::vector<std::tuple<T, int32_t, T>> testRoundWithDecFloatData() {
+    std::vector<std::tuple<T, int32_t, T>> data = {
+        {1.122112, 0, 1},
```
Comment (wangxinshuo-bolt) [inline bolt/functions/sparksql/tests/SparkRoundTest.cpp:68]: I suggest adding some test cases for edge cases:       
```
round(-15, -1)  // expect return -20
round(2147483645, -1)  // expect return 2147483650
```
